### PR TITLE
feat: Add Course Modal

### PR DESCRIPTION
**Pull request by tunny17**
## What does this PR do?

This PR Creates the Add course modal and there are two minor subtasks under which includes:
- Editing the Modal component to accomodate a search feature &
- Editing the textArea component to accomodate a search icon

Fixes # ()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Go to the pathway route and click on add course modal on the course page

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary
[https://www.loom.com/share/0d199bbcb9fe49c2b47bbb6fde89afe2?sid=da94c753-412f-4630-a81f-2c5010249166](url)

### CODE REVIEW

File: apps/dashboard/src/lib/components/Form/TextField.svelte
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Form/TextField.svelte]: @tunny17 Please make a video showing the purpose of this change. Ideally I would love for such a change not be within the generic component used in multiple place just because you need to add a feature to only one place

Comment (rotimi-best) [reply apps/dashboard/src/lib/components/Form/TextField.svelte]: This component has now become more complex

File: apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte]: Why is the name of the variable `group` when the type is `PathwayCourse[]`? What do you think is a better name?

File: apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte]: Why does this begin with a capital letter?

File: apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte]: Why did you change this from camel to snake case?

**Review by rotimi-best — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte
Comment (tunjiadeyemi) [reply apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte]: i kept looking at it and i thought you'd prefer me to ensure consistency with the types, so i used the same variable names but i'll change it back then

File: apps/dashboard/src/lib/components/Form/TextField.svelte
Comment (tunjiadeyemi) [reply apps/dashboard/src/lib/components/Form/TextField.svelte]: https://www.loom.com/share/5b657a4ff58340e9af86eb3d14f6de52?sid=a123e8d6-a1ee-48cf-a45c-ce967f90818a


i really only added it to match the design i saw, all we need to do to use the search option with the modal is this:
  showSearchBar={true} (this is false by default)
  bind:value={searchValue}

File: apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte
Comment (tunjiadeyemi) [reply apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte]: fixed

File: apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte
Comment (tunjiadeyemi) [reply apps/dashboard/src/lib/components/Pathways/components/AddCourseModal.svelte]: fixed

File: apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte
Comment (tunjiadeyemi) [reply apps/dashboard/src/lib/components/Pathways/components/CourseCard.svelte]: but we shouldn't worry about this component anymore, i deleted it and reused the existing card component instead, i think you should review this PR (https://github.com/rotimi-best/classroomio/pull/410) first so after you approve it, i can merge the latest updates into these opened PRs.
this is the PR i made the routing change and as a result of our discussion i reused some components we used for courses like this

File: apps/dashboard/src/lib/components/Form/TextField.svelte
Comment (rotimi-best) [reply apps/dashboard/src/lib/components/Form/TextField.svelte]: I understand, you should probably have just passed a `slot` into that part instead of adding more props and logic into the component. I will refactor it myself

File: apps/dashboard/src/lib/components/Courses/components/Card/index.svelte
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Courses/components/Card/index.svelte]: Why are you doing this?

Is this the way courses work?

File: apps/dashboard/src/lib/components/Form/TextField.svelte
Comment (rotimi-best) [reply apps/dashboard/src/lib/components/Form/TextField.svelte]: The idea I am trying to convey to you is that the search and text field are 2 different behaviours. When updating an existing component, you should think - are they related to each other? The TextField is a dumb component that does one thing, allow users input something.

Anything extra is not a textfield, so in this case, toggling between a search and textfield should be a completely different component managing that UX.

**Review by rotimi-best — CHANGES_REQUESTED** [review]

**Pull request merged**
